apps/api/R/workflows.R

```diff
@@ -229,3 +244,82 @@ getWorkflowFile <- function(req, id, filename, res){
     return(bin)
   }
 }
+
+
+#################################################################################################
+#' Get the list of files in a workflow specified by the id
+#' @param id Workflow id (character)
+#' @return List of files
+#' @author Nihar Sanda
+#* @serializer contentType list(type="application/octet-stream")
+#* @get /<id>/files
+
+getWorkflowFileDetails <- function(req, id, res){
+  Workflow <- tbl(global_db_pool, "workflows") %>%
+    select(id, user_id) %>%
+    filter(id == !!id)
+  
+  qry_res <- Workflow %>% collect()
+  
+  if (nrow(qry_res) == 0) {
+    res$status <- 404
+    return(list(error="Workflow with specified ID was not found"))
+  }
+  else {
+    file_names <- list()
+    file_names <- list.files(paste0(Sys.getenv("DATA_DIR", "/data/"), "workflows/PEcAn_", id))
```

Comment:
If you're querying the database to see if a workflow exists, why are you then making very strong (and often false) assumptions about where the workflow folder is rather than just looking at the folder that's stored in BETY?